tests/cacl/test_cacl_application.py

```diff
@@ -835,6 +835,83 @@ def generate_scale_rules(duthost, ip_type):
     duthost.command("iptables -I INPUT 3 -p tcp -m tcp --dport 22 -j ACCEPT")
 
 
+def verify_cacl_show_acl_rule(duthost):
+    """
+    Converts the CACL rules in the generated SCALE_ACL_FILE into a dict,
+    and verifies that they match the applied acl rules through the `show acl
+    rule` command.
+    """
+
+    acl_json = duthost.command(f"cat {SCALE_ACL_FILE}")["stdout_lines"]
+    acl_rules_expected_dict = json.loads("".join(acl_json))
+
+    acl_sets_extracted = acl_rules_expected_dict["acl"].get("acl-sets", {}).get("acl-set", {})
+    acl_rules_expected = {}
+
+    # Reconstruct the dict such that we can index easily using the actual rules
+    for acl_table, acl_table_conf in acl_sets_extracted.items():
+        # Reformat name as produced by acl-loader
+        acl_table_name = acl_table.upper().replace("-", "_")
+        acl_table_rules = {}
+
+        for acl_entry_num, acl_entry_conf in acl_table_conf['acl-entries']['acl-entry'].items():
+            acl_entry_name = f"RULE_{acl_entry_num}"
+            acl_table_rules[acl_entry_name] = acl_entry_conf
+
+        acl_rules_expected[acl_table_name] = acl_table_rules
+
+    logger.debug(f"Expected rules: {acl_rules_expected}")
+
+    acl_rules_actual = get_cacl_tables_and_rules(duthost)
+
+    logger.debug(f"Actual rules: {acl_rules_actual}")
+
+    incorrect_rules = []
+    incorrect_table_rule_count = []
+
+    # Check each rule in each ACL table on the DUT
+    for acl_table in acl_rules_actual:
```

Comment:
Is it better to use acl_rules_expected to loop? If miss any acl_table_name in acutal output, it doesn't know, right? What do you think?

Reply:
Good point, have fixed - now it loops through expected rules, and checks that actual matches each expected rather than the other way around.